Source/WebKit/UIProcess/Cocoa/ProcessAssertionCocoa.mm

```diff
@@ -389,6 +393,22 @@ - (void)assertion:(RBSAssertion *)assertion didInvalidateWithError:(NSError *)er
     };
 }
 
+ProcessAssertion::ProcessAssertion(AuxiliaryProcessProxy& process, const String& reason, ProcessAssertionType assertionType)
+    : m_assertionType(assertionType)
+    , m_pid(process.processID())
+    , m_reason(reason)
+{
+#if USE(EXTENSIONKIT_ASSERTIONS)
+    NSString *runningBoardAssertionName = runningBoardNameForAssertionType(assertionType);
+    NSString *runningBoardDomain = runningBoardDomainForAssertionType(assertionType);
+    auto capability = adoptNS([_SECapabilities assertionWithDomain:runningBoardDomain name:runningBoardAssertionName]);
+    _SEExtensionProcess *extensionProcess = process.extensionProcess();
+    m_grant = [extensionProcess grantCapabilities:capability.get() error:nil];
```

Comment:
I think it would be useful to log if the `grantCapabilities` operation fails, since we often have bugs about background processes or other things being slowed down or terminated while doing useful things. If this operation fails, we should log it to help debug things in the future.

We should pass an NSError into `error:` and if `m_grant` is `nil` after this, we should log what the error says.